CNAM-441: Create markdown for transformer

Anfor opened the pull request.

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+# Transformers
+
+Transformer is used to turn a dataset from a type `Event` ([Events](Events.md)) 
```
Not necessary one DataSet of Events, it can be one or multiple Datasets.

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+# Transformers
+
+Transformer is used to turn a dataset from a type `Event` ([Events](Events.md)) 
+into another of the same type and another subtype. A  `Transformer` is a class and can accept as a parameter 
```
A class is too strong. You can say that the Transformer accepts a configuration class.

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+into another of the same type and another subtype. A  `Transformer` is a class and can accept as a parameter 
+a configuration class of type
+ [TransformerConfig](https://github.com/X-DataInitiative/SCALPEL-Extraction/blob/master/src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/TransformerConfig.scala).
+This configuration file is a factory to save values and pass them later to the different methods.
```
Weird statement. The configuration is to control the behaviour of the Transformer, and is not per si a Factory. However, it can create an instance of strategy but that is not enough to label it as a factory.

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+This configuration file is a factory to save values and pass them later to the different methods.
+
+
+`transform` is the main method of `Transformer`, this method accepts one or several datasets of type `Event` as input
```
Add 
```markdown
`transform` is the main and the unique public available method of `Transformer`
```

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+
+
+`transform` is the main method of `Transformer`, this method accepts one or several datasets of type `Event` as input
+ and after the different transformations it returns a dataset of type `Event` and a different subtype.
```
Would be better with 
```markdown
And after the applying the logic of the transformation, returns a dataset of type `Event` with a new `category` .
```

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+`transform` is the main method of `Transformer`, this method accepts one or several datasets of type `Event` as input
+ and after the different transformations it returns a dataset of type `Event` and a different subtype.
+ 
+ If we take as example [ExposureTransformer](https://github.com/X-DataInitiative/SCALPEL-Extraction/blob/master/src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/exposures/ExposureTransformer.scala),
```
If we take as __an__ example.

ysebiat commented on Transformer.md:
```diff
@@ -0,0 +1,39 @@
+ and after the different transformations it returns a dataset of type `Event` and a different subtype.
+ 
+ If we take as example [ExposureTransformer](https://github.com/X-DataInitiative/SCALPEL-Extraction/blob/master/src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/exposures/ExposureTransformer.scala),
+ we have as parameter an instance of the class [ExposuresTransformerConfig](https://github.com/X-DataInitiative/SCALPEL-Extraction/blob/master/src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/exposures/ExposuresTransformerConfig.scala)
```
```we pass a as a parameter```

ysebiat commented on Transformer.md:
````diff
@@ -0,0 +1,39 @@
+
+trait TransformerConfig 
+ ```
+The config class stores values and methods.
````
What do you mean by methods?

ysebiat commented on Transformer.md:
````diff
@@ -0,0 +1,39 @@
+class ExposuresTransformerConfig(
+  val exposurePeriodAdder: ExposurePeriodAdder) extends TransformerConfig with Serializable
+ ```  
+The transformer  `ExposureTransformer` takes the config class and used it in the main method.
````
This sentences is redundant and unnecessary.

ysebiat commented on Transformer.md:
````diff
@@ -0,0 +1,39 @@
+      .transform(regulateWithFollowUps(followUps))
+  }
+ ```
+The objective of the `transform` method is to convert the input elements to return the indicated type at the output.
````
Vague phrase.

```markdown
The objective of the `transform` method is thus to combine, filter, check relations between multiples ```Event```s to form a new ```Event```.
In the example of the ```ExposureTransformer```, it combines multiples ```Drug``` ```Event```s based on the logic of the ```exposurePeriodAdder``` of the ```ExposuresTransformerConfig``` to form an ```Exposure``` while making sure that it is contained within the period defined in the ```FollowUp``` ```Event```.

ysebiat reviewed (changes requested):
Some comments on the PR. I think you can elaborate more when explaining what a Transformer stands for, and what we precisely expect from it. 

Also better explain that a config is generated through PureConfig, and thus every parameter is controllable from the exterior.

ysebiat commented on Transformer.md:
```diff
@@ -1,31 +1,30 @@
-`transform` is the main method of `Transformer`, this method accepts one or several datasets of type `Event` as input
- and after the different transformations it returns a dataset of type `Event` and a different subtype.
+`transform` is the main and the unique public available method of `Transformer`, this method accepts one or several datasets  as input
+ and after the applying the logic of the transformation, returns a dataset of type `Event`([Events](Events.md)) with a new `category`.
```
It is either
`and after applying of transformation logic`
Or
`and after the application of the transformation logic`

I prefer the second phrasing.

ysebiat commented on Transformer.md:
````diff
@@ -1,31 +1,30 @@
 trait TransformerConfig 
  ```
-The config class stores values and methods.
+The config class used to control the Transformer's  behaviour.
````
`is used`

ysebiat reviewed (commented):
Needs some tweaks.

ysebiat reviewed (dismissed):
LGTM.

Anfor committed:
CNAM-441: Create markdown for transformer

CNAM-441: Added information to better explain transformer behaviour

CNAM-441: Change sentence to better comprenhension

ysebiat reviewed (approved):
Ready to be merged.
